services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -80,21 +81,39 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)(impl
           config.token
         )
         .map {
-          case Response(HttpStatusCodes.OK, SuccessResponseBody(gitLabFile), _) =>
+          case Response(OK, SuccessResponseBody(gitLabFile), _) =>
             val content = decodeBase64(gitLabFile.content)
             Right(ProjectFile(path, ProjectFileContent(content)))
-          case Response(HttpStatusCodes.OK, FailureResponseBody(error), _) =>
-            Left(
-              HttpException(
-                s"Could not take Project File. Response status: ${HttpStatusCodes.OK}. ResponseBody: $error"
-              )
-            )
+          case Response(OK, FailureResponseBody(error), _) =>
+            Left(HttpException(s"Could not take Project File. Response status: ${OK}. ResponseBody: $error"))
           case Response(responseStatus, _, _) => Left(HttpException(s"Exception. Response status: $responseStatus"))
         }
         .recover { case e: Throwable => Left(HttpException(e.getMessage)) }
     }
   }
 
+  override def deleteFile(
+    project: Project,
+    path: Path,
+    version: Option[PipelineVersion]
+  ): AsyncResult[PipelineVersion] = {
+    val filePath: String = URLEncoderUtils.encode(path.toString)
+    val repositoryId: RepositoryId = project.repositoryId
+    val fileUrl = s"${config.url}projects/${repositoryId.value}/repository/files/$filePath"
+    val name = project.name
+
+    val payload = DeleteFileRequest(s"Deleting file $filePath from repository: $name", config.defaultBranch)
+
+    httpClient.delete[EmptyPayload, DeleteFileRequest](fileUrl, payload = payload, headers = config.token).flatMap {
+      case Response(statusCode, _, _) if statusCode == OK || statusCode == Accepted || statusCode == NoContent =>
+        getUpdatedProjectVersion(project, version).flatMap {
```

Comment:
I'm not sure, need @myazinn approve, but the version should always be None, because we always should get brand new version, otherwise user can send his/her own version to controller and roll back to any version, which is not correct. Also, we don't need to pass Option[PipelineVersion] to this method and to the controller.